### PR TITLE
Support linux-arm

### PR DESCRIPTION
**Pull request by s1341**
Note that the workaround for the r0 being set to 0 by signal return is a bit ugly. We still don't know what exactly the root cause of the issue is, but it's possible that we are hitting this line: http://lxr.free-electrons.com/source/arch/arm/kernel/signal.c?v=3.7#L309

### CODE REVIEW

File: tests/testutil.c [hunk 1]
```diff
@@ -408,7 +408,7 @@ test_util_get_system_module_name (void)
     char * result;
 
     _test_util_system_module_name = malloc (64);
-    p = popen ("find /lib -name \"libc-*.so\" | head -1 | xargs basename"
+    p = popen ("find /lib -name \"libc[-.]*so*\" | head -1 | xargs basename"
```
Comment (oleavr) [inline tests/testutil.c]: Did you check whether this still picks the actual libc and not the symlink on a typical LSB system like Ubuntu?

Comment (s1341) [inline tests/testutil.c]: Seems to work for me on gentoo64 but it appears to fail on ubuntu64... Do you have a suggestion for a better fix?

Comment (oleavr) [inline tests/testutil.c]: How about running the old/current command first, and if that yields nothing we try your glob second?

Comment (s1341) [inline tests/testutil.c]: I just tested two different ubuntu64 machines with the original code. On one the x86_64 libc so was first, and on the other the i386 libc so was first. In other words, the original code wasn't perfect either...

Perhaps we should try something a little more precise?

Comment (oleavr) [inline tests/testutil.c]: Good point! Maybe something like `grep libc /proc/self/maps`?

File: gum/backend-linux/gumprocess-linux.c [hunk 2]
```diff
@@ -1241,7 +1242,8 @@ gum_linux_unparse_ucontext (const GumCpuContext * ctx,
   sc->arm_r7 = ctx->r[7];
   sc->arm_lr = ctx->lr;
 
-  sc->arm_cpsr = (ctx->pc & 1) ? GUM_PSR_THUMB : 0;
+  sc->arm_cpsr = ctx->cpsr;
+  sc->arm_cpsr |= (ctx->pc & 1) ? GUM_PSR_THUMB : 0;
```
Comment (oleavr) [inline gum/backend-linux/gumprocess-linux.c]: I suppose we should clear the thumb bit here if not set?

File: gum/backend-arm/guminterceptor-arm.c [hunk 3]
```diff
@@ -542,7 +542,7 @@ gum_interceptor_backend_write_prolog (GumThumbWriter * tw)
   gum_thumb_writer_put_push_regs (tw, 2, ARM_REG_R0, ARM_REG_R1);
 
   /* reserve space for next_hop */
```
Comment (oleavr) [inline gum/backend-arm/guminterceptor-arm.c]: We should update this comment to avoid confusion.

File: gum/backend-arm/guminterceptor-arm.c [hunk 4]
```diff
@@ -560,7 +560,7 @@ gum_interceptor_backend_write_epilog (GumThumbWriter * tw)
       GUM_FRAME_OFFSET_CPU_CONTEXT + G_STRUCT_OFFSET (GumCpuContext, lr));
 
   /* clear next_hop and low part of GumCpuContext */
```
Comment (oleavr) [inline gum/backend-arm/guminterceptor-arm.c]: We should update this comment to avoid confusion.

**Comment by oleavr** [comment]
Awesome work!

**Pull request merged**